File: TiaCodegen/Commands/Functions/Base/FunctionCall.cs
using System;
using System.Collections.Generic;
using TiaCodegen.Enums;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions.Base
{
    public class FunctionCall : IOperationOrSignal
    {
        public string DebugInfo { get; set; }

        public bool DisableEno { get; set; }

        public string FunctionName { get; set; }

        public string Version { get; set; }

        public string AdditionalInnerXml { get; set; }

        public Dictionary<string, IOperationOrSignalDirectionWrapper> Interface { get; set; }

        public FunctionCall(string functionName, IOperationOrSignal eno = null)
        {
            FunctionName = functionName;
            Interface = new Dictionary<string, IOperationOrSignalDirectionWrapper>();
            Children = new List<IOperationOrSignal>();
            ENO = eno;
            if (eno != null)
            {
                Interface["eno"] = new IOperationOrSignalDirectionWrapper(eno, Direction.Eno);
            }
        }

        public string AdditionalSafetyTemplateValues { get; set; }

        public virtual List<IOperationOrSignal> Children { get; set; }

        public IOperationOrSignal ENO { get; set; }

        public ulong OperationId { get; set; }

        public IOperationOrSignal Parent { get; set; }

        public bool DoNotCreateContact { get; set; }

        public int Cardinality { get; set; }

        public bool HasNoEn { get; set; }

        public virtual int CreateContactAndFillCardinality(IOperationOrSignal parent)
        {
            this.Cardinality = 1;
            return this.Cardinality;
        }

        public override string ToString()
        {
            return this.GetType().Name + "(" + FunctionName + ")";
        }

        public IOperationOrSignal Clone()
        {
            throw new Exception("Not yet supported");
        }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/Base/FunctionCall.ts
import { IOperationOrSignal } from '../../../Interfaces/IOperationOrSignal.js';
import { IOperationOrSignalDirectionWrapper } from '../../../Interfaces/IOperationOrSignalDirectionWrapper.js';
import { Direction } from '../../../Enums/Direction.js';

export class FunctionCall implements IOperationOrSignal {
    safetyTemplateString: string | null = null;
    debugInfo: string | null = null;
    disableEno: boolean = false;
    functionName: string;
    version: string | null = null;
    additionalInnerXml: string | null = null;
    /** Interface map - using plain object to preserve insertion order */
    iface: Record<string, IOperationOrSignalDirectionWrapper>;
    additionalSafetyTemplateValues: string | null = null;
    children: IOperationOrSignal[];
    eno: IOperationOrSignal | null;
    operationId: number = 0;
    parent: IOperationOrSignal | null = null;
    doNotCreateContact: boolean = false;
    cardinality: number = 0;
    hasNoEn: boolean = false;

    constructor(functionName: string, eno: IOperationOrSignal | null = null) {
        this.functionName = functionName;
        this.iface = {};
        this.children = [];
        this.eno = eno;
        if (eno !== null) {
            this.iface['eno'] = new IOperationOrSignalDirectionWrapper(eno, Direction.Eno);
        }
    }

    createContactAndFillCardinality(_parent: IOperationOrSignal): number {
        this.cardinality = 1;
        return this.cardinality;
    }

    toString(): string {
        return `${this.constructor.name}(${this.functionName})`;
    }

    clone(): IOperationOrSignal {
        throw new Error('Not yet supported');
    }
}
